src/tests/cli_tests.py

```diff
@@ -4062,6 +4063,31 @@ def test_allow_weak_hash(self):
         clear_workfiles()
         shutil.rmtree(RNP2, ignore_errors=True)
 
+    def test_allow_sha1_key_sigs(self):
+        RNP7 = RNPDIR + '7'
+        kring = shutil.copytree(data_path(KEYRING_DIR_7), RNP7)
+
+        src, sig = reg_workfiles('cleartext', '.txt', '.sig')
+        random_text(src, 120)
+
+        # rnpkeys, force weak hashes for key generation
+        ret, _, err = run_proc(RNP, ['--homedir', kring, '--password=', '--sign', src, '--output', sig])
+        self.assertNotEqual(ret, 0)
+        self.assertRegex(err, r'(?s)^.*Failed to add signature.*')
+        ret, _, err = run_proc(RNP, ['--homedir', kring, '--password=', '--sign', src, '--output', sig, '--allow-sha1-key-sigs'])
+        self.assertEqual(ret, 0)
+        remove_files(sig)
+
+        ret, _, err = run_proc(RNP, ['--homedir', kring, '--password=', '--sign', src, '--output', sig, '--hash', 'SHA1'])
+        self.assertNotEqual(ret, 0)
+        self.assertRegex(err, r'(?s)^.*Hash algorithm \'SHA1\' is cryptographically weak!.*Weak hash algorithm detected. Pass --allow-weak-hash option if you really want to use it\..*')
+        ret, _, err = run_proc(RNP, ['--homedir', kring, '--password=', '--sign', src, '--output', sig, '--hash', 'SHA1', '--allow-sha1-key-sigs'])
+        self.assertEqual(ret, 0)
+        remove_files(sig)
```

Comment:
You should also try the following scenario:
- import single public key with sha1 signature
- make sure that `rnpkeys --list-keys` lists it by default as invalid
- make sure that `rnpkeys --list-keys --allow-sha1-key-sigs` now lists it as valid
- make sure that encrypting to that key doesn't work by default and works fine with `--allow-sha1-key-sigs`.